owncloudApp/src/androidTest/java/com/owncloud/android/settings/camerauploads/OCSettingsCameraUploadsTest.kt

```diff
@@ -0,0 +1,255 @@
+        onView(withText(R.string.common_yes)).perform(click());
+        //Asserts
+        Assert.assertFalse(mPrefCameraPictureUploads.isChecked)
+        onView(withText(R.string.prefs_camera_picture_upload_path_title)).check(doesNotExist())
+        onView(withText(R.string.camera_picture_upload_on_wifi)).check(doesNotExist())
+    }
+
+    @Test
+    fun disablePictureUploadsRefuse() {
+        enablePictureSubOptions()
+        onView(withText(R.string.prefs_camera_picture_upload)).perform(click());
+        onView(withText(R.string.common_no)).perform(click());
+        //Asserts
+        assertTrue(mPrefCameraPictureUploads.isChecked)
+        //Reset
+        removePictureSubOptions()
+    }
+
+    @Test
+    fun disableVideoUploadsAccept() {
+        enableVideoSubOptions()
+        onView(withText(R.string.prefs_camera_video_upload)).perform(click());
+        onView(withText(R.string.common_yes)).perform(click());
+        //Asserts
+        Assert.assertFalse(mPrefCameraVideoUploads.isChecked)
+        onView(withText(R.string.prefs_camera_video_upload_path_title)).check(doesNotExist())
+        onView(withText(R.string.camera_video_upload_on_wifi)).check(doesNotExist())
+    }
+
+    @Test
+    fun disableVideoUploadsRefuse() {
+        enableVideoSubOptions()
+        onView(withText(R.string.prefs_camera_video_upload)).perform(click());
+        onView(withText(R.string.common_no)).perform(click());
+        //Asserts
+        assertTrue(mPrefCameraVideoUploads.isChecked)
+        //Reset
+        removeVideoSubOptions()
+    }
+
+    @Test
+    fun cameraFolderView() {
+        enablePictureSubOptions()
+        //Asserts
+        onView(withText(String.format(
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title),
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title_required))))
+            .check(matches(isDisplayed()))
+        //Reset
+        removePictureSubOptions()
+    }
+
+    @Test
+    fun cameraOpenPicker() {
+        enablePictureSubOptions()
+        val cameraFolder = Environment.getExternalStoragePublicDirectory(
+            Environment.DIRECTORY_DCIM
+        ).absolutePath + "/Camera";
+        Intents.init()
+        //Asserts
+        onView(withText(String.format(
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title),
+            activityRule.activity.getString(R.string.prefs_camera_upload_source_path_title_required))))
+            .perform(click())
+        Intents.intended(IntentMatchers.hasComponent(LocalFolderPickerActivity::class.java.name))
+        IntentMatchers.hasExtra(LocalFolderPickerActivity.EXTRA_PATH, cameraFolder)
+        Intents.release()
+        onView(withText(android.R.string.cancel)).perform(click())
+        //Reset
+        removePictureSubOptions()
+    }
+
+    @Test
+    fun switchOriginalFileWillBe() {
```

Comment:
Method not used

Reply:
This one is a @Test

Reply:
Ups, you are right 

[<img src="https://user-images.githubusercontent.com/15261363/57386676-2416bd00-71b5-11e9-99d3-b43878c550b8.gif" width="350px"/>](https://user-images.githubusercontent.com/15261363/57386676-2416bd00-71b5-11e9-99d3-b43878c550b8.gif)